cincinnati/src/lib.rs

```diff
@@ -157,6 +160,35 @@ impl Graph {
             .try_fold((), |_, (from, to)| self.add_edge(&from, &to))
     }
 
+    /// Returns tuples of ReleaseId and its version String for releases for which
+    /// filter_fn returns true.
+    ///
+    /// filter_fn is able to mutate the release as it receives a mutable borrow.
+    /// It will only get passed a ConcreteRelease or AbstractRelease for each
+    /// release.
+    pub fn find_by_fn_mut<F>(&mut self, mut filter_fn: F) -> Vec<(ReleaseId, String)>
+    where
+        F: FnMut(Option<&mut ConcreteRelease>, Option<&mut AbstractRelease>) -> bool,
```

Comment:
This is a bit of a strange signature. Unless you explicitly need to handle the None/None and Some/Some cases, it would be better to pass the actual enum.

Reply:
It's indeed a bit strange. I wanted to prevent exposing the enum but if you'd prefer that I'm fine going that route.

Reply:
I changed the signature to use the enum and it's actually not as terrible as I thought it would be.